src/style/fog.js

```diff
@@ -90,14 +88,16 @@ class Fog extends Evented {
     _transitionable: Transitionable<Props>;
     _transitioning: Transitioning<Props>;
     properties: PossiblyEvaluated<Props>;
-    sampler: FogSampler;
 
     constructor(fogOptions?: FogSpecification) {
         super();
         this._transitionable = new Transitionable(properties);
         this.set(fogOptions);
         this._transitioning = this._transitionable.untransitioned();
-        this.sampler = new FogSampler();
+    }
+
+    get sampler(): FogSampler {
+        return new FogSampler(this.properties.get('range'));
```

Comment:
Should we have a way of not creating a new FogSampler in a get accessor? Since this is called from here as well.

https://github.com/mapbox/mapbox-gl-js/blob/747e70077b0cf0540edcf1665e6265d450670c76/src/ui/map.js#L2185

Which creates a new one for each call to `getFogOpacity`.

My personal preference is having getters be no-ops dead accessors, and otherwise `createSampler` for new instancing.